packages/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java

```diff
@@ -117,8 +124,27 @@ protected void onCreate(Bundle savedInstanceState) {
       String type = intent.getStringExtra("type");
       if (channel != null) {
```

Comment:
I ran your code and this channel check will make the opening fail when `channel != null`.
Steps to reproduce:
1) Install and open the app normally.
2) Minimize the app by tapping in the home button.
3) Use the app shortcut.
4) Exit the app using the back button. ## Not the home button.
5) Use the app shortcut.  ## "nothing happens".

Possible fix that is not a breaking change:
In `ShortcutHandlerActivity` class :
```
      // Get the Intent that started this activity and extract the string
      Intent intent = getIntent();
      String type = intent.getStringExtra("type");
      startActivity(getIntentToOpenMainActivity(this, type));

      finish();
```
``` 
        launchIntentForPackage.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        launchIntentForPackage.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
```

Reply:
Yes, I know about this issue and it isn't something I wanted to fix in my PR. If you can, please, check the current version of the plug-in and make sure that the bug was there before my fix. I don't know if I should fix it in this PR, maybe you can open an issue or even submit a PR? Or do you think that it's better to fix the bug in this PR? Anyway, I will check your solution, because I also faced this issue and was thinking about the ways to fix it.

Reply:
@collinjackson ☝️ ? Another PR? It seems small enough to land together with this version.

Reply:
@BugsBunnyBR I know why does this happen. When you click the "Back" button, the activity is being killed. So, the activity is killed now, but `channel` reference and `QuickActionsPlugin` class instance is still alive. So, later, when you click on the shortcut, it tries to send a message to the channel, because it is not null. But as you know, the view is already detached, and you'll get only this message in logcat `FlutterView.send called on a detached view, channel=`. Unfortunately, you cannot put a callback there to react on this error case and, maybe, start the activity again, because of API design. I've opened an issue a week ago https://github.com/flutter/flutter/issues/35628 but maybe it is possible to fix without changes in `FlutterNativeView` API design...

Reply:
Yeah. I know how to solve the problem. Just remove the static for the channel. In general, having static fields that are associated with context are a bad ideia. But removing the static makes that we need some other few changes, like listening to onNewIntent, that is why I think another pr will be needed to have a proper handling.